R/merge.R

```diff
@@ -21,8 +21,8 @@ merge.data.table = function(x, y, by = NULL, by.x = NULL, by.y = NULL, all = FAL
   if (!missing(by) && !missing(by.x))
     warning("Supplied both `by` and `by.x/by.y`. `by` argument will be ignored.")
   if (!is.null(by.x)) {
-    if ( !is.character(by.x) || !is.character(by.y))
-      stop("A non-empty vector of column names are required for `by.x` and `by.y`.")
+    if (length(by.x) == 0L || !is.character(by.x) || !is.character(by.y))
```

Comment:
aha! I was just looking at this code yesterday and something looked funny but I didn't bother stress testing it. nice catch!